cookbook/common/parent.mk

```diff
@@ -14,64 +14,75 @@ fast_serialize:
 	@for dir in $(SUBDIRS) ; do \
 		echo "processing ${PWD}/$$dir"; \
 		trimmed=$${dir%/}; \
-		test -f $$dir/Makefile && \
-		PREFIX=$$trimmed $(MAKE) fast_serialize; \
+	    if [[ -f $$dir/Makefile ]]; then \
+    		PREFIX=$$trimmed $(MAKE) fast_serialize || exit; \
+		fi; \
 	done
 
 .PHONY: fast_register
 fast_register: ## Registers new code changes using the last built image (assumes current HEAD refers to a built image).
 	@for dir in $(SUBDIRS) ; do \
 		echo "processing ${PWD}/$$dir"; \
 		trimmed=$${dir%/}; \
-		test -f $$dir/Makefile && \
-		PREFIX=$$trimmed $(MAKE) fast_register; \
+	    if [[ -f $$dir/Makefile ]]; then \
+    		PREFIX=$$trimmed $(MAKE) fast_register || exit; \
+		fi; \
 	done
 
 .PHONY: register
 register: ## Builds, pushes and registers all docker images, workflows and tasks in all sub directories.
 	@for dir in $(SUBDIRS) ; do \
 		echo "processing ${PWD}/$$dir"; \
-		test -f $$dir/Makefile && \
-		$(MAKE) -C $$dir register; \
+	    if [[ -f $$dir/Makefile ]]; then \
+    		$(MAKE) -C $$dir register || exit; \
+		fi; \
 	done
 
 .PHONY: serialize
 serialize: ## Builds and serializes all docker images, workflows and tasks in all sub directories.
 	@for dir in $(SUBDIRS) ; do \
 		echo "processing ${PWD}/$$dir"; \
-		test -f $$dir/Makefile && \
-		$(MAKE) -C $$dir serialize; \
+		echo "$$dir"; \
+	    if [[ -f $$dir/Makefile ]]; then \
+	        if [[ "$$dir" != 'docs/' ]]; then \
```

Comment:
we don't need to check for this in the other targets?  register, fast register, etc